_blogposts/2023-04-15-improving-interop.mdx

```diff
@@ -0,0 +1,383 @@
+---
+author: rescript-team
+date: "2023-04-15"
+title: Better interop with customizable variants
+badge: roadmap
+description: |
+  A tour of new capabilities coming in ReScript v11
+---
+
+ReScript v11 is around the corner, and it comes packed with new features that will improve interop with JavaScript/TypeScript. We will shine some light on each significant feature in a series of blog posts.
+
+Interop with JavaScript/TypeScript is a top priority for ReScript. Recently we've made some changes to the runtime representation of variants that'll allow you to use variants for a large number of new interop scenarios, zero cost. This is important, because variants is _the_ feature of ReScript, enabling great data modeling, pattern matching and more.
+
+_Pattern matching_ is really the key here. The changes we've made will enable you to write idiomatic ReScript leveraging pattern matching in many more scenarios.
+
+These changes will ship in `v11` of ReScript, and in this post we'll tell you all you need to know about them.
+
+## TLDR;
+
+This is a long post, so here's a summary for those of you who just want the broad strokes.
+
+### Customizable runtime representation
+
+We're making the runtime representation of variants customizable. This will allow you to cleanly map variants to external data and APIs in many more cases than before.
+
+### Zero cost bindings to discriminated unions
+
+Variants with inline records will map cleanly to JavaScript/TypeScript [discriminated unions](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions).
+
+Discriminated unions are getting more and more popular in TypeScript (for good reason!) and this will open up a whole new venue for simple interop for ReScript. It will let you effortlessly utilize the power of ReScript's pattern matching and similar features, without needing any runtime conversion.
+
+In addition it'll also greatly improve debuggability, because the runtime representation will map cleanly to the variant type you've defined.
+
+### Unboxed (untagged) variants
+
+We also introduce untagged variants - variants where the underlying runtime representation can be a primitive, without a specific discriminator. This will let you cleanly map to things like heterogenous array items, nullable values, and more.
+
+Now let's dive into the details.
+
+## The current state
```

Comment:
You are already mention in the examples below how we do X in the old system. This section talks too much of the past.

The new examples highlight really well how interop issues are now solved.

Alternatively colocate parts of its content to the relevant feature sections to keep the context if needed.